Source/PyFR/ColorTable.cu

```diff
@@ -1,17 +1,17 @@
 
 #include "ColorTable.h"
 
-static ::thrust::system::cuda::vector< Color> PaletteVec;
-static ::thrust::system::cuda::vector< float> PivotsVec;
-static std::vector<Color> RuntimePalette;
-static std::vector<float> RuntimePivots;
-
-RuntimeColorTable::RuntimeColorTable(FPType cmin, FPType cmax,
-  const std::vector<Color>& palette,
-  const std::vector<float>& pivots)
+static ::thrust::system::cuda::vector< Color> PaletteVec[2];
+static ::thrust::system::cuda::vector< float> PivotsVec[2];
+static std::vector<Color> RuntimePalette[2];
+static std::vector<float> RuntimePivots[2];
+
+RuntimeColorTable::RuntimeColorTable(int pipeline, FPType cmin, FPType cmax,
+  const std::vector<Color>& palette, const std::vector<float>& pivots)
```

Comment:
The problem is that multiple copies of RuntimeColorTable is constructed ( one per contour iso-value ) on the host and they should all share the same state. On top of that, because of the way the coloring was implemented, these objects also need to have a valid copy constructor for the device. Before the runtime color-table was added this wasn't a problem, as everything was stack based member variables.

So if we tried to add member varaibles for ```thrust::...::vector``` the compilation fails as that object doesn't have copy constructors that are valid in both host and device.  So the work around is to use only raw pointers.

But if we tried to malloc / free gpu pointers for each RuntimeColorTable instance we would get double free when doing the host side copies ( so now we need ref counting? )

The quick and dirty way out was to simply make the arrays static so the memories lifetime was independent of the color table.

The long term solution is to redesign how we are doing coloring. The current solution is 'nice' since it allows the contours to write out as uchar4 instead of float, but because they are designed in a 'stateless' manner it causes problems. If I had to revisit the problem I would rewrite the ```MapFieldOntoIsosurfaces``` methods and have them be passed a color table from the filter itself.